server/channels/app/platform/web_hub.go

```diff
@@ -620,30 +632,51 @@ func areAllInactive(conns []*WebConn) bool {
 // It requires 3 functionalities which need to be very fast:
 // - check if a connection exists or not.
 // - get all connections for a given userID.
+// - get all connections for a given channelID.
 // - get all connections.
 type hubConnectionIndex struct {
 	// byUserId stores the list of connections for a given userID
 	byUserId map[string][]*WebConn
+	// byChannelID stores the list of connections for a given channelID.
+	byChannelID map[string][]*WebConn
 	// byConnection serves the dual purpose of storing the index of the webconn
 	// in the value of byUserId map, and also to get all connections.
 	byConnection   map[*WebConn]int
 	byConnectionId map[string]*WebConn
 	// staleThreshold is the limit beyond which inactive connections
 	// will be deleted.
 	staleThreshold time.Duration
+
+	store  store.Store
+	logger mlog.LoggerIFace
 }
 
-func newHubConnectionIndex(interval time.Duration) *hubConnectionIndex {
+func newHubConnectionIndex(interval time.Duration,
+	store store.Store,
+	logger mlog.LoggerIFace,
+) *hubConnectionIndex {
 	return &hubConnectionIndex{
 		byUserId:       make(map[string][]*WebConn),
+		byChannelID:    make(map[string][]*WebConn),
 		byConnection:   make(map[*WebConn]int),
 		byConnectionId: make(map[string]*WebConn),
 		staleThreshold: interval,
+		store:          store,
+		logger:         logger,
 	}
 }
 
 func (i *hubConnectionIndex) Add(wc *WebConn) {
 	i.byUserId[wc.UserId] = append(i.byUserId[wc.UserId], wc)
+
+	cm, err := i.getChannelMembersForUser(wc.UserId)
+	if err != nil {
+		panic(fmt.Errorf("error while getting GetAllChannelMembersForUser: %v", err))
+	}
+	for chID := range cm {
+		i.byChannelID[chID] = append(i.byChannelID[chID], wc)
```

Comment:
Mostly a curiosity of mine: do we have heap profiles around to understand how heavy this map can get?

Reply:
Very minimal. 4.25MB. From the latest test.
[heap_200K.zip](https://github.com/user-attachments/files/17537000/heap_200K.zip)